### ISSUE TITLE
Desync, possibly caused by deleted 1:1 chat

### ISSUE THREAD

**Issue by Valodim** [tulir/mautrix-telegram#221]
```
Sep 06 22:45:26 metiadoc python[18120]: [2018-09-06 22:45:26,133] [ERROR@mau.user] Failed to handle Telegram update
Sep 06 22:45:26 metiadoc python[18120]: Traceback (most recent call last):
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 875, in ensure_joined
Sep 06 22:45:26 metiadoc python[18120]:     await self._join_room_direct(room_id)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 136, in _send
Sep 06 22:45:26 metiadoc python[18120]:     errcode=errcode, message=message)
Sep 06 22:45:26 metiadoc python[18120]: mautrix_appservice.errors.MatrixRequestError: 403: {"errcode":"M_FORBIDDEN","error":"You are not invited to this room."}
Sep 06 22:45:26 metiadoc python[18120]: During handling of the above exception, another exception occurred:
Sep 06 22:45:26 metiadoc python[18120]: Traceback (most recent call last):
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 875, in ensure_joined
Sep 06 22:45:26 metiadoc python[18120]:     await self._join_room_direct(room_id)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 136, in _send
Sep 06 22:45:26 metiadoc python[18120]:     errcode=errcode, message=message)
Sep 06 22:45:26 metiadoc python[18120]: mautrix_appservice.errors.MatrixRequestError: 403: {"errcode":"M_FORBIDDEN","error":"You are not invited to this room."}
Sep 06 22:45:26 metiadoc python[18120]: During handling of the above exception, another exception occurred:
Sep 06 22:45:26 metiadoc python[18120]: Traceback (most recent call last):
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_telegram/abstract_user.py", line 133, in _update_catc
Sep 06 22:45:26 metiadoc python[18120]:     await self._update(update)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_telegram/abstract_user.py", line 189, in _update
Sep 06 22:45:26 metiadoc python[18120]:     await self.update_message(update)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_telegram/abstract_user.py", line 370, in update_messa
Sep 06 22:45:26 metiadoc python[18120]:     return await portal.handle_telegram_message(self, sender, update)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_telegram/portal.py", line 1436, in handle_telegram_me
Sep 06 22:45:26 metiadoc python[18120]:     response = await self.handle_telegram_text(source, intent, is_bot, evt)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_telegram/portal.py", line 1335, in handle_telegram_te
Sep 06 22:45:26 metiadoc python[18120]:     await intent.set_typing(self.mxid, is_typing=False)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 642, in set_typing
Sep 06 22:45:26 metiadoc python[18120]:     await self.ensure_joined(room_id)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 881, in ensure_joined
Sep 06 22:45:26 metiadoc python[18120]:     await self.bot.invite(room_id, self.mxid)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 514, in invite
Sep 06 22:45:26 metiadoc python[18120]:     await self.ensure_joined(room_id)
Sep 06 22:45:26 metiadoc python[18120]:   File "/home/matrix/matrix-appservice-tg/lib/python3.5/site-packages/mautrix_appservice/intent_api.py", line 879, in ensure_joined
Sep 06 22:45:26 metiadoc python[18120]:     raise IntentError("Failed to join room {} as {}".format((room_id), (self.mxid)), e)
Sep 06 22:45:26 metiadoc python[18120]: mautrix_appservice.errors.IntentError: Failed to join room !pyRfCjxLSbegBGFqJp:stratum0.org as @telegrambot:stratum0.org
```

We just ended up in a weird situation. I'm not fully clear on the details, but here's roughly what we did:
- User A is logged in with tg and matrix, User B only with tg
- Users A and B have had a tg chat open (since before logging in)
- User A deletes tg chat in tg
- User B writes msg in this deleted tg chat in tg

This caused the following situation:
- User A is invited into an mx room X, which has User B's puppet account in it and the message
- Users A and B are invited into a *different* mx room (this is !pyRfC... above) Y, with no puppeted users (and not the bot) in it. Writing in this room.

Now:
- If User A writes in mx room X, the message is mirrored to tg and room Y
- If User A writes in the tg chat, it shows up in room X and Y
- If User B writes in the tg chat, it shows up only in Y
- If User A or B write in mx room Y,  the messages show up in mx but are not mirrored to tg

Bottom line, the rooms the users end up in aren't particularly predictable and useful for communication :)

**Comment by Valodim** [comment]
Closing as stale
